pkg/api/v1/resource/helpers.go

```diff
@@ -192,6 +247,12 @@ func ExtractContainerResourceValue(fs *v1.ResourceFieldSelector, container *v1.C
 		return convertResourceMemoryToString(container.Resources.Requests.Memory(), divisor)
 	case "requests.ephemeral-storage":
 		return convertResourceEphemeralStorageToString(container.Resources.Requests.StorageEphemeral(), divisor)
+	case "resourcesAllocated.cpu":
+		return convertResourceCPUToString(container.ResourcesAllocated.Cpu(), divisor)
+	case "resourcesAllocated.memory":
+		return convertResourceMemoryToString(container.ResourcesAllocated.Memory(), divisor)
+	case "resourcesAllocated.ephemeral-storage":
+		return convertResourceEphemeralStorageToString(container.ResourcesAllocated.StorageEphemeral(), divisor)
```

Comment:
this extends allowed values for the downward API in ways old kubelets won't understand, right?

Reply:
It looks like older kubelet should fail validation if users try to expose these fields in their containers (a check I need to update to allow these fields).

End result is that pod won't create, so we won't get to the point where kubelet gets bad info - atleast for 1.18 this is true, I didn't try 1.17, but same validation code exists there. Is this an issue? This seems like the right behavior.

Reply:
Are you concerned about the case where apiserver has been upgraded, but kubelet has yet to be? I that event, the old kubelet will not recognize these new fields that apiserver lets through - it results in container start error. So not great UX - sometimes it might work and sometimes it won't, until upgrade process complete. 

Is there a way we can handle this?